packages/svelte/src/internal/client/render.js

```diff
@@ -2616,73 +2621,14 @@ export function spread_props(props) {
  * @returns {Exports & { $destroy: () => void; $set: (props: Partial<Props>) => void; }}
  */
 export function createRoot(component, options) {
```

Comment:
With this change `createRoot` is basically == `mount`. I'm wondering if we should therefore should just get rid of `createRoot` and move the `$set`/`$destroy` interface onto `mount` (in a follow-up PR)

Reply:
Another option is this:

```js
import { createProps, createRoot } from 'svelte';
import App from './App.svelte';

const props = createProps({ count: 1 });
const destroy = createRoot(App, { props });

// later
props.count += 1;
destroy();
```

In the case where you didn't need to update the props after creation, it would just be this:

```js
import { createRoot } from 'svelte';
import App from './App.svelte';

const destroy = createRoot(App, { props: { count: 1 } });

// later
destroy();
```

To me this feels a bit more harmonious than splodging together props with `$set` and `$destroy` methods. And it would render `accessors` unnecessary.

(Obviously we'd still keep that stuff around in compat mode.)

Reply:
There's more than accessors, the returned instance can also contain functions etc that are exposed through
```svelte
<script>
  export function x() {..}
</script>
```
so we won't be getting away with just `destroy`.
I also think that getting an object back you can call `$set` and `$destroy` on feels like a better API than creating a props object you interact with separately.
Lastly, if it was only `mount`, we could introduce a corresponding `hydrate`, with the difference being that `mount` never sets the hydration flags, which if done right would hopefully enable Rollup etc to treeshake the whole hydration code away, ensuring smaller builds for (for example) web component people.